packages/core/test/scuttle.spec.js

```diff
@@ -0,0 +1,126 @@
+) {
+  return intrinsics.some((intrinsic) => {
+    try {
+      const _ = globalRef[intrinsic]
+      return false
+    } catch (e) {
+      if (err(intrinsic) === e.message) {
+        return true
+      }
+      throw new Error('Unknown error thrown at testScuttling: ' + e.message)
+    }
+  })
+}
+
+test('scuttle - no opts', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+})
+
+test('scuttle - opts as bool', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, false)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, true)
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - opts as object', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: false })
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: true })
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - exceptions', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {
+    enabled: true,
+    exceptions: ['/[a-zA-Z0-8]*Array/', 'RegExp'],
+  })
+  t.is(
+    testScuttling(vmGlobalThis, ['String']),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+})
+
+test('scuttle - scuttle func', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  let globalRef
+  const cb = (realm, scuttle) => ((globalRef = realm), scuttle(realm))
+  t.throws(
+    (_) => scuttle(vmGlobalThis, { enabled: true, scuttlerName: 'SCUTTLER' }),
+    SCUTTLER_NAME_ERROR
+  )
+  Object.defineProperty(vmGlobalThis, 'SCUTTLER', { value: cb })
+  scuttle(vmGlobalThis, { enabled: true, scuttlerName: 'SCUTTLER' })
+  t.is(
+    globalRef,
+    vmGlobalThis,
+    'expect global reference to be the same as provided by the scuttler function'
+  )
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - resilient', (t) => {
```

Comment:
(style) I try to do a single assertion per test so that every assertion is guaranteed to be executed

Reply:
Can you give an example?  What's a "step"?

Reply:
I think 1 assertion per test is a nice idea to strive for, but I've seen it become counterproductive if applied despite difficulties. So let's try, but if there's a price to pay, consider if it's worth it.

Reply:
yes, certainly if the test has an expensive setup, there's little upside in repeatedly spending it

Reply:
fixed (addressed by https://github.com/LavaMoat/LavaMoat/pull/758#discussion_r1469769937)